### ISSUE TITLE
Pretrain data not found in AllaVA

### ISSUE THREAD

**Issue by lucasjinreal** [dvlab-research/MiniGemini#20]
Hi, the pretrained data used allava images both from laion nad vfan.

But the laion part image names are totally different from ALLava's images format.

I tried to found:

```
465440.jpeg
320609
```

they all used in minigemini_pretrain.json but can not be found in ALLava images folder.

```
ls -f images | grep 465440
46544031.jpeg
(base) ➜  allava_laion git:(main) ✗ ls -f images | grep 320609     
132060956.jpeg
43206091.jpeg
```

why is that？

**Comment by lucasjinreal** [comment]
How to mapping this image file really?

![image](https://github.com/dvlab-research/MiniGemini/assets/21303438/98c1b195-ba6c-4444-a89f-4cb631b1351f)

**Comment by daicver** [comment]
I've also noticed this issue. It seems that after the existing allava was updated, some pictures were deleted. Can the repository owner take a look at the latest allava dataset and provide relevant information? @yanwei-li

**Comment by lucasjinreal** [comment]
It's not deleted, the index number doesn't really match the ALLava's images, besides, I think the minigemini data provided image name are trucked, where is `jpeg` suffix? This really shouldn't happen for them, make users very confused and missleading..

**Comment by daicver** [comment]
I try to use the url of minigemini pretrain dataset to match the url of allava dataset, minigemini pretrain dataset is missing about 6939 images.

**Comment by lucasjinreal** [comment]
Missing is normal, the question is, we have to using url to match the correct image id (filename)

this is rediculous.

**Comment by g-h-chen** [comment]
@daicver @lucasjinreal Hi, thanks for using the ALLaVA data. I am from the ALLaVA group. We did have a silent update soon after we release the data. And it seems that the Mini-Gemini project was using the data before the update. 
In the original version, the `images` entry looked like `allava_laion/allava_laion_512763` and all image filenames were **without a suffix** either, which means they are mapped correctly.
In the current version, we made simultaneous adjustments to annotations and image filenames **with suffix added**. 
We will fix this issue soon with the Mini-Gemini team. Stay tuned!

**Comment by g-h-chen** [comment]
>So, looks like minigemini were using older data, Just wonder, does the gpt repsonse also changed?

NO change in GPT-4V response.

>Is the newer data is a super collection to older one?

No. In short, we only add postfixes to image filenames and annotation files so that one can preview it easily. 

>we can just mapping the name with url propabaly?

Sure you can do so, but we have uploaded the images in our repo as well which saves some effort for you.

**Comment by lucasjinreal** [comment]
Yes, I downloaded the images, here is a mapping script to adjust minigemini images back to new Allava image filename:

```python
import json
import sys

minig = sys.argv[1]
allava = sys.argv[2]

# Load JSON file a
with open(minig, 'r') as f:
    data_a = json.load(f)

# Load JSON file b
with open(allava, 'r') as f:
    data_b = json.load(f)

# Convert list of dictionaries in b to a dictionary for easy lookup
dict_b = {item['url']: item['image'] for item in data_b}

# Iterate over items in a, replace 'image' value with value from b
for item in data_a:
    if 'url' in item.keys() and item['url'] in dict_b:
        item['image'] = dict_b[item['url']]

# Save the modified data back to a.json
with open('a_fix_gemini.json', 'w') as f:
    json.dump(data_a, f, indent=2)
```
For anyone to use directly.

**Comment by lucasjinreal** [comment]
Oh, I found using url to mapping ,still get some file unable to map correclty. Any solution?

For example:


```
/ALLaVA-4V/allava_laion/images/281029 not found

/allava_laion/images/387904
```

**Comment by yanwei-li** [comment]
Hi, we @g-h-chen are working together to align the data and will update the data file soon, please stay tuned.

**Comment by lucasjinreal** [comment]
Looks like the truth is not exaclty said as @g-h-chen , the new Allava actually **delete** some images which minigemini used.

**Comment by yanwei-li** [comment]
Hi, @lucasjinreal @daicver we have updated the ALLaVA data in our files, please download them in the original link [Mini-Gemini-Pretrain](https://huggingface.co/datasets/YanweiLi/Mini-Gemini-Pretrain) and [Mini-Gemini-Instruction](https://huggingface.co/datasets/YanweiLi/Mini-Gemini-Instruction). We will also re-train our model to find the effect of data change.

**Comment by daicver** [comment]
ok, thanks

**Comment by g-h-chen** [comment]
Dudes, (hopefully) a final comment here:
1. The `id` entry of each item for `allava_laion` (caption and instruction) is **not** unique. **The number of samples is 505588, and the number of unique ids is 484532**. However, the contents are not the same for the samples sharing the same id. The reason is that when ALLaVA project started, we tried out some prompting strategies which led to our final version, but we accidentally re-generated those samples when performing large-scale distillation. However, we kept those samples anyway considering the cost. 

2. Mini-Gemini team and us have updated the aligned data. Sorry for the inconvenience caused!
